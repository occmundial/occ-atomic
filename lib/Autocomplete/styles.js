import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
