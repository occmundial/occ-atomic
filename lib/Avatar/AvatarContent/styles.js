import fonts from '../../tokens/fonts.json';
import colors from '../../tokens/colors.json';
import borderRadius from '../../tokens/borderRadius.json';
import legacyFont from '../../subatomic/fonts';

const { avatar, text, border } = colors;
const transition = 'all cubic-bezier(0.25,0.46,0.45,0.94) 0.2s';

const font = fonts['body-regular-strong'];

export default {
	wrap: {
		position: 'relative',
		width: '100%',
		height: '100%'
	},
	cont: {
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		borderRadius: borderRadius['br-full'],
		overflow: 'hidden',
		background: avatar.bg.default,
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center'
	},
	capital: {
		display: 'inline-block',
		font: `${font.fontWeight} ${font.fontSize}/${font.lineHeight} ${legacyFont.body}`,
		color: text.indigo.primary,
		gridRowStart: '1',
		gridColumnStart: '1',
		textAlign: 'center'
	},
	disabled: {
		pointerEvents: 'none',
		background: avatar.bg.disabled,
		'& $camera': {
			opacity: 1
		}
	},
	disabledCapital: {
		color: text.corp.disabled
	  },
	clickable: {
		transition,
			'&:hover': {
				background: avatar.bg.hover
			},
			'&:active': {
				background: avatar.bg.active
			}
	  },
	editable: {
		'&:hover $camera': {
			opacity: 1
		},
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	},
	person: {
		gridRowStart: 1,
		gridColumnStart: 1,
		transition
	},
	  photo: {
		border: `1px solid ${border.default.subtle}`
	},
	hoverPhoto: {
		'&:hover $overlay': {
			background: avatar.bg.overlay.hover
		},
		'&:active $overlay': {
			background: avatar.bg.overlay.active
		}
	  },
	camera: {
		gridRowStart: 1,
		gridColumnStart: 1,
		opacity: 0,
		transition
	},
	overlay: {
		borderRadius: '50%',
		position: 'absolute',
		top: '0',
		left: '0',
		width: '100%',
		height: '100%',
		display: 'grid',
		alignItems: 'center',
		justifyContent: 'center',
		transition,
		border: `1px solid transparent`
	  }
};
