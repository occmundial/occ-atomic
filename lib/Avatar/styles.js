import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
