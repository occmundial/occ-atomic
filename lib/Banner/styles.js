import spacing from '../subatomic/spacing';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

export default {
  container: { padding: spacing.small, backgroundColor: colors.ink },
  textBanner: { display: 'inline-block' },
  textFlex: {
    textAlign: 'center',
    [`@media screen and (max-width:${grid.xs}px)`]: { textAlign: 'start' },
  },
  iconFlex: { [`@media screen and (max-width:${grid.xs}px)`]: { padding: spacing.small } },
  icon: { cursor: 'pointer' },
};
