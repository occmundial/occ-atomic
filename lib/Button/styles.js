import legacyColors from '../subatomic/colors';
import legacyFont from '../subatomic/fonts';
import icons from '../subatomic/icons';

import colors from '../tokens/colors.json';
import spacing from '../tokens/spacing.json';
import borderRadius from '../tokens/borderRadius.json';
import fonts from '../tokens/fonts.json';
import shadows from '../tokens/shadows.json';
import { legacyIcons } from '../subatomic/mappedIcons';

const { sec, secDark, grey500, secLight } = legacyColors;

const sm = fonts['button-small'];
const md = fonts['button-medium'];
const lg = fonts['button-large'];

const renderIcon = (props, color) => {
    if (!legacyIcons[props.icon]) return '';
    if (props.icon) {
        return icons.base(icons[props.icon].icon(color));
    } else if (props.iconRight) {
        return icons.base(icons[props.iconRight].icon(color));
    }
    return '';
};

export default {
    btn: {
        display: 'inline-block',
        boxSizing: 'border-box',
        position: 'relative',
        maxWidth: '100%',
        marginBottom: 0,
        border: 0,
        padding: [spacing['size-3'], spacing['size-4']],
        borderRadius: borderRadius['br-xs'],
        font: `${sm.fontWeight} ${sm.fontSize}/${sm.lineHeight} ${legacyFont.body}`,
        textAlign: 'center',
        textDecoration: 'none',
        whiteSpace: 'nowrap',
        transition: '0.3s all',
        cursor: 'pointer',
        userSelect: 'none',
        touchAction: 'manipulation',
        '&:focus': {
            outline: 'none'
        },
        '&:hover': {
            textDecoration: 'none'
        }
    },
    cont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        pointerEvents: 'none'
    },
    loadCont: {
        display: 'flex',
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 0,
        left: 0,
        pointerEvents: 'none'
    },
    // Themes
    primary: {
        background: colors.button.primary.bg.default,
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.primary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.primary.bg.hover
        },
        '&:active': {
            background: colors.button.primary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.primary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-pink']
        },
        '&$disabled': {
            background: colors.button.primary.bg.disabled,
            color: colors.text.white.secondary,
            cursor: 'not-allowed'
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    secondary: {
        background: colors.button.secondary.bg.default,
        color: colors.text.indigo.primary,
        '&:hover': {
            background: colors.button.secondary.bg.hover
        },
        '&:active': {
            background: colors.button.secondary.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '&$disabled': {
            background: colors.button.secondary.bg.disabled,
            color: colors.text.indigo.secondary
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiary: {
        background: 'transparent',
        color: colors.text.indigo.primary,
        outline: `2px solid ${colors.button.tertiary.border.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-indigo']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.brand.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.indigo.secondary,
            outline: `2px solid ${colors.button.tertiary.border.disabled}`
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.brand.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.brand.default]))
        }
    },
    tertiaryWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        outline: `2px solid ${colors.button.tertiary.border.inverse.default}`,
        outlineOffset: '-2px',
        '&:hover': {
            background: colors.button.tertiary.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.tertiary.bg.inverse.active
        },
        '&:focus': {
            outline: `2px solid ${colors.button.tertiary.border.inverse.default}`
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    ghostPink: {
        background: 'transparent',
        color: sec,
        border: 'none',
        paddingLeft: [0, '!important'],
        paddingRight: [0, '!important'],
        '&:hover, &:active': {
            color: secDark,
            '& span span': {
                background: props => renderIcon(props, [secDark])
            },
        },
        '& span span': {
            background: props => renderIcon(props, [sec])
        },
        '&$disabled': {
            background: `transparent !important`,
            color: `${secLight} !important`,
            border: `none !important`,
            '& span span': {
                background: props => renderIcon(props, [grey500])
            }
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([sec]))
        }
    },
    ghostGrey: {
        background: 'transparent',
        color: colors.text.corp.secondary,
        '&:hover': {
            background: colors.button.ghost.bg.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-corp']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.default.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.corp.disabled
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.default.disabled])
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.icon.default.default]))
        }
    },
    ghostWhite: {
        background: 'transparent',
        color: colors.text.white.primary,
        '&:hover': {
            background: colors.button.ghost.bg.inverse.hover
        },
        '&:active': {
            background: colors.button.ghost.bg.inverse.active
        },
        '&:not(:active):focus-visible': {
            boxShadow: shadows['focus-white']
        },
        '& span span': {
            background: props => renderIcon(props, [colors.icon.inverse.default])
        },
        '&$disabled': {
            background: 'transparent',
            color: colors.text.white.secondary
        },
        '&$disabled span span': {
            background: props => renderIcon(props, [colors.icon.inverse.disabled])
        }
    },
    loading: {
        pointerEvents: 'none',
        '& $cont': {
            visibility: 'hidden'
        }
    },
    disabled: {
        cursor: 'default',
        pointerEvents: 'none',
    },
    // Sizes
    md: {
        font: `${md.fontWeight} ${md.fontSize}/${md.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-5']],
        '&$iconOnly': {
            padding: [spacing['size-2'], spacing['size-2']]
        }
    },
    lg: {
        font: `${lg.fontWeight} ${lg.fontSize}/${lg.lineHeight} ${legacyFont.body}`,
        padding: [spacing['size-3'], spacing['size-6']],
        '&$iconOnly': {
            padding: [spacing['size-3'], spacing['size-3']]
        }
    },
    // Block
    block: {
        display: 'block',
        width: '100%'
    },
    // Icon
    icon: {
        marginRight: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconRight: {
        width: 16,
        height: 16,
        marginLeft: spacing['size-2'],
        display: 'inline-block',
        transition: '0.3s all'
    },
    iconOnly: {
        padding: [spacing['size-2'], spacing['size-2']],
        '& span span, & svg': {
            marginRight: 0
        }
    },
    loadIcon: {
        width: 24,
        height: 24,
        fill: colors.icon.inverse.default
    },
    round: {
        borderRadius: '50%'
    }
};
