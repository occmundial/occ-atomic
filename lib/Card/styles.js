import colors from '../subatomic/colors';
import shadows from '../subatomic/shadows';
import spacing from '../subatomic/spacing';
import grid from '../subatomic/grid';

const styles = {
    card: {
        background: colors.white,
        borderRadius: 6,
        padding: spacing.small,
        position: 'relative',
        transition: '0.3s box-shadow, 0.3s transform',
        [`@media screen and (min-width:${grid.sm}px)`]: {
            padding: spacing.medium
        }
    },
    cardNoPadding: {
        extend: 'card',
        padding: 0,
        [`@media screen and (min-width:${grid.sm}px)`]: {
            padding: 0
        }
    },
    flat: {
        boxShadow: shadows.lvl1,
        zIndex: 'auto'
    },
    rest: {
        boxShadow: shadows.lvl0,
        zIndex: 'auto'
    },
    raised: {
        boxShadow: shadows.lvl4,
        zIndex: 4
    },
    raisable: {
        '&:hover': {
            boxShadow: shadows.lvl4,
            transform: `translateY(-${spacing.xTiny}px)`,
            zIndex: 4
        }
    }
};

export default styles;
