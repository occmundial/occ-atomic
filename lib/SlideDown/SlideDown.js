import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';
import Icon from '../Icon';
import Tag from '../Tag';
import Flexbox from '../Flexbox';

import colors from '../tokens/colors.json';

import usePrevious from '../Provider/usePrevious';

const textSizes = {
  sm: { small: true },
  md: { standard: true },
  lg: { extraLarge: true },
  default: { standard: true }
};

const _themeColors = {
  blue: {
    icon: colors.icon.brand.default,
    text: {
      primary: true
    }
  },
  default: {
    icon: colors.icon.default.bold,
    text: {
      highEmphasis: true
    }
  }
};

/** Component to create an accordion section. The content will be hidden by default, and shown after click. */
function SlideDown({
  expanded,
  classes,
  title,
  children,
  tag,
  textSize,
  strong,
  theme,
  noJustified,
  onToggle,
  icon,
  divider,
  noPadding,
  testId
}) {
  const [_expanded, setExpanded] = useState(!!expanded);
  const prevExpanded = usePrevious(!!expanded);
  const themeColors = _themeColors[theme] || _themeColors['default'];

  const toggleContent = e => {
    e.stopPropagation();
    setExpanded(!_expanded);

    if (onToggle) onToggle(!_expanded);
  };

  const getTextProps = () => ({
    ...(textSizes[textSize] || textSizes.default),
    ...themeColors.text
  });

  useEffect(() => {
    if (prevExpanded !== expanded) {
      setExpanded(expanded);
    }
  }, [prevExpanded, expanded]);

  return (
    <div className={classes.wrapper}>
      <div
        className={`${classes.button} ${
          textSize === 'lg' ? classes.largePadding : classes.normalPadding
        }${
          !noPadding ? ` ${classes.buttonPadding}` : ''
        }`}
        role="button"
        onClick={toggleContent}
        data-testid={testId}
      >
        <Flexbox
          display="flex"
          justifyContent={!noJustified ? 'between' : null}
          alignItems="center"
          className={classes.flexContainer}
        >
          <Flexbox display="flex" alignItems="center" wrap="wrap">
            {icon ? (
              <Icon
                iconName={icon}
                width={16}
                height={16}
                colors={[themeColors.icon]}
                className={classes.leftIcon}
              />
            ) : (
              ''
            )}
            <Text
              {...getTextProps()}
              strong={strong}
              tag="label"
              className={classes.text}
            >
              {title}
            </Text>
            {tag && (
              <Tag theme="info" className={classes.tag}>
                {tag}
              </Tag>
            )}
          </Flexbox>
          <Icon
            iconName="arrowDown"
            colors={[themeColors.icon]}
            className={`${classes.icon}${
              _expanded ? ` ${classes.rotateIcon}` : ''
            }`}
            width={16}
            height={16}
            testId="slidedown__arrow-down"
          />
        </Flexbox>
      </div>
      <div
        className={`${classes.contentWrapper}${
          _expanded ? ` ${classes.showContentWrapper}` : ''
        }`}
      >
        <div
          className={`${classes.content}${
            _expanded ? ` ${classes.showContent}` : ''
          }`}
        >
          {children}
        </div>
      </div>
      {divider ? <div className={classes.divider} /> : ''}
    </div>
  );
}

SlideDown.defaultProps = {
  expanded: false,
  textSize: 'md',
  strong: false,
  theme: 'default',
  noJustified: false,
  icon: '',
  noPadding: false
};

SlideDown.propTypes = {
  classes: PropTypes.object,
  children: PropTypes.node,
  /** Title of the tab. */
  title: PropTypes.string,
  /** Set this as true to show the content by default. If the external prop is the same of the internal state the component dont toggle */
  expanded: PropTypes.bool,
  /** Text to show as a label right next to the Title */
  tag: PropTypes.string,
  /** Function to call when the SlideDown expands or collapses. It returns true if it is expanded and false if it is collapsed. */
  onToggle: PropTypes.func,
  /** Size of the Text to show sm, md, lg*/
  textSize: PropTypes.oneOf(['sm', 'md', 'lg']),
  /** If the text is required to be bold*/
  strong: PropTypes.bool,
  /** Theme of the SlideDown */
  theme: PropTypes.oneOf(['default', 'blue']),
  /** The Title container has an alignment by default. Use this property if you need to remove it. */
  noJustified: PropTypes.bool,
  /** Shows a divider at the bottom*/
  divider: PropTypes.bool,
  /** Use the name of any of the icons available in the library. */
  icon: PropTypes.string,
  /** The container has by default a side padding. Use this property if you need to remove it. */
  noPadding: PropTypes.bool,
  /** The testId property adds the data attribute data-testid to the main element and should be used for testing only. */
  testId: PropTypes.string,
};

export default SlideDown;
