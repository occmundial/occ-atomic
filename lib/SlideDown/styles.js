import spacing from '../tokens/spacing.json';
import colors from '../tokens/colors.json';

export default {
  wrapper: {
    marginBottom: spacing['size-4']
  },
  button: {
    cursor: 'pointer',
  },
  normalPadding: {
    paddingTop: spacing['size-3'],
    paddingBottom: spacing['size-3']
  },
  largePadding: {
    paddingTop: spacing['size-4'],
    paddingBottom: spacing['size-4']
  },
  buttonPadding: {
    paddingLeft: spacing['size-4'],
    paddingRight: spacing['size-4']
  },
  text: {
    cursor: 'pointer'
  },
  tag: {
    marginTop: spacing['size-1'],
    marginLeft: spacing['size-2'],
    pointerEvents: 'none'
  },
  icon: {
    transition: 'rotate cubic-bezier(0.25,0.46,0.45,0.94) 0.2s'
  },
  leftIcon: {
    marginRight: spacing['size-2']
  },
  rotateIcon: {
    transform: 'rotate(-180deg)'
  },
  content: {
    overflow: 'hidden',
    transform: 'translateY(-10px)',
    transition: 'transform cubic-bezier(0.25,0.46,0.45,0.94) 0.2s'
  },
  showContent: {
    transform: 'translateY(0px)',
    transition: 'transform cubic-bezier(0.25,0.46,0.45,0.94) 0.2s 0.05s',
    marginBottom: spacing['size-3']
  },
  contentWrapper: {
    display: 'grid',
    gridTemplateRows: '0fr',
    transition:
      'grid-template-rows cubic-bezier(0.25,0.46,0.45,0.94) 0.2s 0.05s, opacity cubic-bezier(0.25,0.46,0.45,0.94) 0.2s',
    opacity: 0
  },
  showContentWrapper: {
    opacity: 1,
    gridTemplateRows: '1fr',
    transition:
      'grid-template-rows cubic-bezier(0.25,0.46,0.45,0.94) 0.2s, opacity cubic-bezier(0.25,0.46,0.45,0.94) 0.2s 0.05s'
  },
  divider: {
    border: `1px solid ${colors.border.default.subtle}`,
    marginTop: `-${spacing['size-0']}`
  },
  flexContainer: {
    gap: spacing['size-4']
  }
};
