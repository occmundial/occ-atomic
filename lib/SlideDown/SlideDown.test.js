import React from 'react';
import { shallow, mount } from 'enzyme';
import SlideDown from './SlideDown';
import styles from './styles';
import SlideDownJSS from './index';
import Text from '../Text';
import Icon from '../Icon';
import colors from '../tokens/colors.json';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe('SlideDown', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<SlideDown classes={classes} title="Title" />);
		expect(wrapper).toMatchSnapshot();
	});

	it('shows the title', () => {
		const wrapper = mount(<SlideDown title="Title" classes={classes} />);
		expect(wrapper.find('label').at(0).text()).toEqual('Title');
	});

	it('rotates the arrow icon', () => {
		const wrapper = mount(
			<SlideDown expanded classes={classes} title="Title" testId="slidedown-test"/>
		);
		
		expect(wrapper.find('Icon .rotateIcon').length).toBe(1)
		wrapper.find({ "data-testid": 'slidedown-test'}).simulate('click', {
			stopPropagation: () => {}
		});
		expect(wrapper.find('Icon .rotateIcon').length).toBe(0)
	});

	it('shows a Tag on the title', () => {
		const wrapper = mount(<SlideDown classes={classes} title="Title" />);
		wrapper.setProps({ tag: 'Tag text' });
		expect(wrapper.find('label').at(1).text()).toBe('Tag text');
	});

	it('renders the child content', () => {
		const wrapper = shallow(
			<SlideDown classes={classes} title="Title" testId="slidedown-test">
				Children content
			</SlideDown>
		);
		expect(wrapper.find('.showContent').length).toBe(0);

		wrapper.find({ "data-testid": 'slidedown-test'}).simulate('click', {
			stopPropagation: () => {}
		});

		expect(wrapper.find('.showContent').length).toBe(1);
		expect(wrapper.find('.showContent').text()).toEqual('Children content');
	});

	it('Text have the right properties', () => {
		const wrapper = shallow(
			<SlideDown classes={classes} textSize={'lg'} title="Title">
				Children content
			</SlideDown>
		);
		expect(wrapper.find(Text).props().extraLarge).toEqual(true);
	});

	it('Text and Icon has theme color', () => {
		const wrapper = shallow(
			<SlideDown classes={classes} title="Title">
				Children content
			</SlideDown>
		);
		expect(wrapper.find(Icon).props().colors).toEqual([colors.icon.default.bold]);
		expect(wrapper.find(Text).props().primary).toBeFalsy();
		wrapper.setProps({ theme: 'blue' });
		expect(wrapper.find(Text).props().primary).toEqual(true);
		expect(wrapper.find(Icon).props().colors).toEqual([colors.icon.brand.default]);
	});

	it('render a custom title', () => {
		const wrapper = mount(
			<SlideDown
				customTitle={
					<div data-testid="custom-title" style={{ display: 'flex', gap: '24px', alignItems: 'center' }}>
						<p style={{ margin: 0 }}>
							Custom <b>SlideDown</b> Title
						</p>
						<Icon iconName="search" width={16} height={16} />
					</div>
				}
				title="This title should not render"
				classes={classes}>
				Children content
			</SlideDown>
		);
		expect(wrapper.find({ "data-testid": 'custom-title'}).exists()).toEqual(true)
		expect(wrapper.find('label').exists()).toEqual(false)
	});
});

describe('SlideDownJSS', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<SlideDownJSS />);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('SlideDown styles', () => {
	it('matches the snapshot', () => {
		expect(styles).toMatchSnapshot();
	});
});
