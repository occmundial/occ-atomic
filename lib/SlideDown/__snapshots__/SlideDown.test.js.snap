// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideDown matches the snapshot 1`] = `ShallowWrapper {}`;

exports[`SlideDown styles matches the snapshot 1`] = `
Object {
  "button": Object {
    "cursor": "pointer",
  },
  "buttonPadding": Object {
    "paddingLeft": "16px",
    "paddingRight": "16px",
  },
  "content": Object {
    "overflow": "hidden",
    "transform": "translateY(-10px)",
    "transition": "transform cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
  },
  "contentWrapper": Object {
    "display": "grid",
    "gridTemplateRows": "0fr",
    "opacity": 0,
    "transition": "grid-template-rows cubic-bezier(0.25,0.46,0.45,0.94) 0.2s 0.05s, opacity cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
  },
  "disabled": Object {
    "cursor": "not-allowed",
  },
  "divider": Object {
    "boxShadow": "inset 0 -1px 0px 0px #EDEDF1",
  },
  "flexContainer": Object {
    "gap": "16px",
  },
  "icon": Object {
    "transition": "rotate cubic-bezier(0.25,0.46,0.45,0.94) 0.2s",
  },
  "largePadding": Object {
    "paddingBottom": "16px",
    "paddingTop": "16px",
  },
  "leftIcon": Object {
    "marginRight": "8px",
  },
  "normalPadding": Object {
    "paddingBottom": "12px",
    "paddingTop": "12px",
  },
  "rotateIcon": Object {
    "transform": "rotate(-180deg)",
  },
  "showContent": Object {
    "marginBottom": "12px",
    "transform": "translateY(0px)",
    "transition": "transform cubic-bezier(0.25,0.46,0.45,0.94) 0.2s 0.05s",
  },
  "showContentWrapper": Object {
    "gridTemplateRows": "1fr",
    "opacity": 1,
    "transition": "grid-template-rows cubic-bezier(0.25,0.46,0.45,0.94) 0.2s, opacity cubic-bezier(0.25,0.46,0.45,0.94) 0.2s 0.05s",
  },
  "tag": Object {
    "marginLeft": "8px",
    "pointerEvents": "none",
  },
  "text": Object {
    "cursor": "pointer",
  },
  "wrapper": Object {},
}
`;

exports[`SlideDownJSS matches the snapshot 1`] = `ShallowWrapper {}`;
